plugins/utils_operators/file_operators.py

```diff
@@ -22,19 +23,38 @@ class DownloadFileOperator(BaseOperator):
     @apply_defaults
     def __init__(
         self,
-        file_url: str,
-        dir: str,
-        filename: str,
+        file_url: str = None,
+        file_url_task_id: str = None,
+        file_url_task_key: str= None,
+
+        dir: str = None,
+        dir_task_id: str = None,
+        dir_task_key: str= None,
+        
+        filename: str = None,
+        filename_task_id: str = None, 
+        filename_task_key: str= None,
+
         overwrite_if_exists: bool = True,
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
-        self.file_url = file_url
-        self.dir = dir
-        self.filename = filename
+        # Assigning these vars in single lines to emphasize that they are alternatives to one another
+        self.file_url, self.file_url_task_id, self.file_url_task_key = file_url, file_url_task_id, file_url_task_key
+        self.dir, self.dir_task_id, self.dir_task_key = dir, dir_task_id, dir_task_key
+        self.filename, self.filename_task_id, self.filename_task_key = filename, filename_task_id, filename_task_key
+                    
```

Comment:
Similar comment as the `GetOrCreateResourceOperator` regarding parameters. What is the expected behaviour when, say, all parameters are passed when _either_ of them is expected? Is this behaviour clear to devs? And, is it an issue? If so, should we add an assert check at the initialization stage of the operator?

Reply:
Yeah - take a look at my response here:
https://github.com/open-data-toronto/operations/pull/18/files#r681963499

Reply:
Future PR